torch/csrc/jit/mobile/module.cpp

```diff
@@ -52,7 +56,9 @@ void set_train_recurse(
     TORCH_INTERNAL_ASSERT(false, "'training' attribute not found");
   }
   for (const auto& slot : obj->slots()) {
-    if (slot.isObject()) {
+    // slots is a list of IValue. Continue setting training attribute only
+    // if the slot is an object and a module.
+    if (slot.isObject() && slot.toObjectRef().type()->is_module()) {
```

Comment:
After this fix, is it closer to what jit is doing when setting training?

Reply:
Yeah copy their logic here from `torch/jit/module.cpp`:
```
void Module::train(bool on) {
  for (Module m : modules()) {
    if (auto slot = m._ivalue()->type()->findAttributeSlot("training")) {
      m._ivalue()->setSlot(*slot, on);
    } else {
      TORCH_INTERNAL_ASSERT("'training' attribute not found");
    }
  }
}
```

When they iterate all slot and try to get `modules`, this check will be used: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/api/module.h#L508